pkg/controllers/provisioningv2/cluster/controller_test.go

```diff
@@ -15,3 +25,250 @@ func TestRegexp(t *testing.T) {
 	assert.False(t, mgmtNameRegexp.MatchString("c-12345b"))
 	assert.False(t, mgmtNameRegexp.MatchString("ac-12345b"))
 }
+
+func TestController_generateProvisioningClusterFromLegacyCluster(t *testing.T) {
+	tests := []struct {
+		name    string
+		cluster *v3.Cluster
+	}{
+		{
+			name: "test-default",
+			cluster: &v3.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "c-11111",
+				},
+				Spec: v3.ClusterSpec{
+					ClusterSpecBase:    v3.ClusterSpecBase{},
+					FleetWorkspaceName: "test-fleet-workspace-name",
+				},
+			},
+		},
+		{
+			name: "test-cluster-agent-customization",
+			cluster: &v3.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "c-22222",
+				},
+				Spec: v3.ClusterSpec{
+					ClusterSpecBase: v3.ClusterSpecBase{
+						ClusterAgentDeploymentCustomization: getTestClusterAgentCustomizationV3(),
+						FleetAgentDeploymentCustomization:   getTestFleetAgentCustomizationV3(),
+					},
+					FleetWorkspaceName: "test-fleet-workspace-name",
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			h := handler{}
+
+			obj, _, err := h.generateProvisioningClusterFromLegacyCluster(tt.cluster, tt.cluster.Status)
+			provCluster := obj[0].DeepCopyObject().(*v1.Cluster)
```

Comment:
Question: why do we need to deep copy the cluster object?  
(same question for the one in the TestController_createNewCluster function)